frictionless/validate/main.py

```diff
@@ -26,6 +26,11 @@ def validate(source=None, type=None, **options):
         type = "package" if file.multipart else file.type
         if type == "table":
             type = "resource"
+        if "descriptor" in options:
+            file = system.create_file(
+                options.get("descriptor"), basepath=options.get("basepath", "")
+            )
+            type = file.type
```

Comment:
I would fall back to `resource` type if the descriptor is detected to be a `table` as `file.type == 'table'`. Otherwise, something like this `validate(descriptor='table.csv')` will crash the lib (we might need to add a test for this edge case)

Reply:
I have added a fall back to resource if the type is not a package.

Furthermore, I would like to mention that passing a ```table``` type file as a ```descriptor``` like - ```validate(descriptor='table.csv')``` won't raise any exception but will retrieve an error: 
```Python
print(validate(descriptor='table.csv').flatten())
[1, None, None, 'resource-error']
```